### ISSUE TITLE
[terra-table] Screen Reader is not reading the table header name.

### ISSUE THREAD

**Issue by saket2403** [cerner/terra-core#3321]
# Bug Report

## Description
Screen Reader is not reading the table header name. On tabbing to navigate through the table while using screen reader, it skips reading the table header name.

## Expected Behavior
Screen reader should read the table name also.

## Environment

* Component Name and Version:
* Browser Name and Version:
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile):

**Comment by dkasper-was-taken** [comment]
@saket2403 Can you provide more detail, perhaps a screenshot or code example from the dev-site to point towards which text element should be being read by the screenreader?

**Comment by saket2403** [comment]
I have updated the issue to be a spike on adding caption element as discussed with dave.
